memtable.hh

```diff
@@ -272,8 +272,8 @@ public:
     const db::replay_position& replay_position() const {
         return _replay_position;
     }
-    const db::rp_set& rp_set() const {
-        return _rp_set;
+    db::rp_set rp_set() {
+        return std::exchange(_rp_set, {});
```

Comment:
Nitpick (optional): this is a surprising implementation for a getter method. I wonder if it could have been made less dangerous with a name change (e.g., get_and_clear_rp_set), by adding "&&" before the "{" , or a comment.

Reply:
True. I get overly non-modifying when dealing with headers like memtable. But this should be renamed. Fixed.